### PR TITLE
[ETFE-3115] Add submission model creation and update declaration view/controller

### PR DESCRIPTION
**Pull request by MJCallahanPage**
Note: raised as draft as #22 needs to go into main first and then this branch rebased from main.

### CODE REVIEW

**Comment by LewisMitchell** [comment]
Rebased from #22

File: app/models/submitChangeDestination/UpdateEadEsadModel.scala [hunk 1]
```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
```
Comment (LewisMitchell) [inline app/models/submitChangeDestination/UpdateEadEsadModel.scala:2]:
```suggestion
 * Copyright 2024 HM Revenue & Customs
```

File: conf/app.routes [hunk 2]
```diff
@@ -37,4 +37,5 @@ POST        /trader/:ern/movement/:arc/declaration            controllers.Declar
 ->          /                                                 exportInformation.Routes
 ->          /                                                 destination.Routes
 ->          /                                                 guarantor.Routes
+->          /                                                 movement.Routes
```
Comment (LewisMitchell) [inline conf/app.routes:40]: This can be removed (duplicate routes, most likely from my rebase, sorry!)

File: conf/messages [hunk 3]
```diff
@@ -191,6 +196,11 @@ taskList.section.transport.journeyType = Journey type
 taskList.section.transport.transportArranger = Transport arranger
 taskList.section.transport.firstTransporter = First transporter
 taskList.section.transport.units = Transport units
+taskList.section.items = Items
+taskList.section.items.items = Items
+taskList.section.documents = Documents
+taskList.section.documents.sad = Single Administrative Documents
+taskList.section.documents.documents = Documents
```
Comment (LewisMitchell) [inline conf/messages:203]: Probably can be removed (side effect of my rebase)

File: conf/messages [hunk 4]
```diff
@@ -168,7 +168,12 @@ signedOut.guidance.notSaved = We have not saved your answers.
 
 activeTrader.link = Change trader type
 
+lrn.subHeading = Change destination for {0}
+
+taskList.title = Draft movement
 taskList.lrn = Local Reference Number (LRN): {0}
+taskList.consignee = Consignee: {0}
+taskList.notProvided = Not provided
```
Comment (LewisMitchell) [inline conf/messages:176]: Probably can be removed (except taskList.lrn - side effect of my rebase)

File: test/pages/sections/consignor/ConsignorSectionSpec.scala [hunk 5]
```diff
@@ -28,11 +28,14 @@ class ConsignorSectionSpec extends SpecBase {
         implicit val dr: DataRequest[_] = dataRequest(FakeRequest(), emptyUserAnswers.set(ConsignorAddressPage, testUserAddress))
         ConsignorSection.isCompleted mustBe true
       }
+    }
 
+    "should return true" - {
```
Comment (LewisMitchell) [inline test/pages/sections/consignor/ConsignorSectionSpec.scala:33]: Can be combined with the above (probably a rebase issue)

**Review by LewisMitchell — CHANGES_REQUESTED** [review]

File: app/models/submitChangeDestination/DestinationChangedModel.scala [hunk 6]
```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
```
Comment (LewisMitchell) [inline app/models/submitChangeDestination/DestinationChangedModel.scala:2]:
```suggestion
 * Copyright 2024 HM Revenue & Customs
```

File: test/models/submitChangeDestination/DestinationChangedModelSpec.scala [hunk 7]
```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
```
Comment (LewisMitchell) [inline test/models/submitChangeDestination/DestinationChangedModelSpec.scala:2]:
```suggestion
 * Copyright 2024 HM Revenue & Customs
```

File: test/models/submitChangeDestination/UpdateEadEsadModelSpec.scala [hunk 8]
```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
```
Comment (LewisMitchell) [inline test/models/submitChangeDestination/UpdateEadEsadModelSpec.scala:2]:
```suggestion
 * Copyright 2024 HM Revenue & Customs
```

File: app/models/submitChangeDestination/UpdateEadEsadModel.scala [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+
+  def apply(implicit request: DataRequest[_]): UpdateEadEsadModel = {
+
+    val journeyTimeValue: JourneyTime = (request.userAnswers.get(JourneyTimeHoursPage), request.userAnswers.get(JourneyTimeDaysPage)) match {
+      case (Some(hours), _) => Hours(hours.toString)
+      case (_, Some(days)) => Days(days.toString)
+      case _ =>
+        logger.error("Missing mandatory UserAnswer for journeyTime")
+        throw MissingMandatoryPage("Missing mandatory UserAnswer for journeyTime")
+    }
```
Comment (LewisMitchell) [inline app/models/submitChangeDestination/UpdateEadEsadModel.scala:54]: One thing to check is that if the user has 'X hours' set in 801 and switches it to 'X days' how does that affect the submission?

Comment (LewisMitchell) [reply app/models/submitChangeDestination/UpdateEadEsadModel.scala:54]: Could we just make a `getFromUserAnswersOnly` call here instead

Comment (LewisMitchell) [reply app/models/submitChangeDestination/UpdateEadEsadModel.scala:54]: Then if there is no 'local' answer for either page, set the journey time to None?

Comment (MJCallahanPage) [reply app/models/submitChangeDestination/UpdateEadEsadModel.scala:54]: Yeah, that's a good spot. I think we can use `getFromUserAnswersOnly` and do what you've suggested around setting it to `None` if neither exist. That seems like the best option.

File: test/models/submitChangeDestination/UpdateEadEsadModelSpec.scala [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+
+      implicit val dr: DataRequest[_] = dataRequest(
+        request = fakeRequest,
+        answers = emptyUserAnswers.set(JourneyTimeHoursPage, 1)
+      )
+
+      UpdateEadEsadModel.journeyTimeValue mustBe Some(Hours("1"))
+    }
+
+    "must return Some(Days) when hours" in {
```
Comment (LewisMitchell) [inline test/models/submitChangeDestination/UpdateEadEsadModelSpec.scala:46]:
```suggestion
    "must return Some(Days) when days" in {
```

Comment (MJCallahanPage) [reply test/models/submitChangeDestination/UpdateEadEsadModelSpec.scala:46]: poop

**Review by LewisMitchell — APPROVED** [review]

**Pull request merged**